src/Avalonia.Visuals/Media/TextFormatting/TextEllipsisHelper.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System.Collections.Generic;
+using Avalonia.Media.TextFormatting.Unicode;
+
+namespace Avalonia.Media.TextFormatting
+{
+    internal class TextEllipsisHelper
+    {
+        public static List<ShapedTextCharacters>? Collapse(TextLine textLine, TextCollapsingProperties properties, bool isWordEllipsis)
+        {
+            var shapedTextRuns = textLine.TextRuns as List<ShapedTextCharacters>;
+
+            if (shapedTextRuns is null)
+            {
+                return null;
+            }
+
+            var runIndex = 0;
+            var currentWidth = 0.0;
+            var collapsedLength = 0;
+            var textRange = textLine.TextRange;
+
+            // TODO: From where this is supposed to come now?
+            var flowDirection = FlowDirection.LeftToRight;
```

Comment:
At this point text runs are not reordered so we always have glyphs in one direction

Reply:
Added one to `TextCollapsingProperties.Collapse`. Let me know if you would rather have a context struct here or if we should pass paragraph properties instead.